### PR TITLE
Add Tera formatter

### PR DESCRIPTION
**Pull request by drwpow**
Adds formatter based on `prettydiff`. Added because I wanted to use Tera templates, but was sad that my HTML wasn’t being auto-formatted 🙂.

It’s not perfect, but it’s workable quality, at least for a first pass (better than some other VS Code formatters, anyway). It’s based on `prettydiff`, which seems to be a very common dependency for other VS Code formatters.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+on: [pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+      - run: npm install
+      - run: npm test
```
Comment (drwpow) [inline .github/workflows/test.yml:12]: This runs `npm test` on every pull request, so test breakages are easy to see.

File: README.md [hunk 2]
```diff
@@ -1,7 +1,6 @@
 # Tera README
 
-Tera Template syntax highlighting. Expect nothing fancy and a lot of bugs from this extension.
-Based on https://github.com/danielchatfield/atom-jinja2
+VS Code extension for syntax highlighting and formatting for [Tera][tera] templates. Based on https://github.com/danielchatfield/atom-jinja2.
```
Comment (drwpow) [inline README.md:3]: I wanted to remove the “expect nothing fancy” line because now it is fancy! 🙂

File: package.json [hunk 3]
```diff
@@ -1,53 +1,71 @@
...
-    "publisher": "karunamurti",
-    "engines": {
-        "vscode": "^1.19.0"
-    },
-    "categories": [
-        "Languages"
-    ],
-    "license": "MIT",
-    "keywords": [
-        "boo"
```
Comment (drwpow) [inline package.json:15]: Added keywords (assuming VS Code searches these)

File: tests/format.test.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
+const { prettifyTera } = require("../lib/format");
+
+describe("Tera formatter", () => {
```
Comment (drwpow) [inline tests/format.test.js:3]: Adds a basic test for the formatting itself.

File: package.json [hunk 5]
```diff
@@ -1,53 +1,76 @@
...
+  "bugs": {
+    "url": "https://github.com/karuna/tera-vscode/issues",
+    "email": "karuna.murti@gmail.com"
+  },
+  "icon": "images/logo.png",
+  "activationEvents": [
+    "onLanguage:tera"
+  ],
+  "scripts": {
+    "vscode:prepublish": "npm run build",
```
Comment (drwpow) [inline package.json:66]: I believe this should prepare the package for VS Code (found in other similar extensions)

File: package.json [hunk 6]
```diff
@@ -1,53 +1,78 @@
...
-    "homepage": "https://github.com/karuna/tera-vscode",
-    "bugs": {
-        "url": "https://github.com/karuna/tera-vscode/issues",
-        "email": "karuna.murti@gmail.com"
-    },
-    "icon": "images/logo.png"
-}
\ No newline at end of file
+        "extensions": [
+          ".tera"
```
Comment (drwpow) [inline package.json:35]: I noticed other packages’ extensions start with `.`

File: package.json [hunk 7]
```diff
@@ -1,53 +1,78 @@
...
+  "displayName": "Tera",
+  "description": "A powerful, easy-to-use template language based on Rust, inspired by Jinja2 and Django Templates.",
+  "version": "0.0.2",
+  "publisher": "karunamurti",
+  "engines": {
+    "vscode": "^1.19.0"
+  },
+  "categories": [
+    "Programming Languages",
+    "Formatters"
```
Comment (drwpow) [inline package.json:12]: Added `Formatters`, and changed it to `Programming Languages` (seems to be the new default)

**Comment by karuna** [comment]
let me check later

**Pull request merged**